src/sentry/tasks/relocation.py

```diff
@@ -95,5 +137,333 @@ def uploading_complete(uuid: str) -> None:
     silo_mode=SiloMode.REGION,
 )
 def preprocessing_scan(uuid: str) -> None:
+    """
+    Performs the very first part of the `PREPROCESSING` step of a `Relocation`, which involves
+    decrypting the user-supplied tarball and picking out some useful information for it. This let's
+    us validate a few things:
+
+        - Ensuring that the user gave us properly encrypted data (was it encrypted? With the right
+          key?).
+        - Ensuring that the org slug the user supplied exists in the provided JSON data.
+        - Recording the slugs of the orgs the relocation is attempting to import.
+        - Recording the usernames of the users the relocation is attempting to import.
+
+    Of the preprocessing tasks, this is the most resource-onerous (what if the importer provides a
+    2GB JSON blob? What if they have 20,000 usernames? Etc...) so we should take care with our retry
+    logic and set careful limits.
+
+    This function is meant to be idempotent, and should be retried with an exponential backoff.
+    """
+
+    relocation: Optional[Relocation]
+    attempts_left: int
+    (relocation, attempts_left) = start_task(
```

Comment:
nit: Can we rename this to `start_relocation_task`? Initially thought it might be generally for starting any celery task

Reply:
Done.